### PR TITLE
Preparation refactoring towards v5.0 (part 3) 

### PR DESCRIPTION
**Pull request by aslesarenko**
- [x] CompilerSettings class added
- [x] added missing ScalaDocs

### CODE REVIEW

File: sigmastate/src/main/scala/sigmastate/interpreter/CryptoFunctions.scala [hunk 1]
```diff
@@ -5,8 +5,12 @@ import scorex.crypto.hash.Blake2b256
 object CryptoFunctions {
   lazy val soundnessBytes: Int = CryptoConstants.soundnessBits / 8
 
+  /** @hotspot don't beautify this code. Used in Interpreter.verify. */
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/interpreter/CryptoFunctions.scala:8]: Please add method desc, please note that it is not so clear now from the code that trimming takes place.

File: sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala [hunk 2]
```diff
@@ -150,6 +149,7 @@ trait Interpreter extends ScorexLogging {
     }
   }
 
+  /** Reduces `exp` to SigmaProp under the default (empty) environment. */
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala:152]: This comment does not sat anything beyond what can be trivially deduced from the code (even with no language knowledge probably), is it needed then ?

File: sigmastate/src/main/scala/sigmastate/lang/SigmaCompiler.scala [hunk 3]
```diff
@@ -11,10 +11,20 @@ import sigmastate.lang.SigmaPredef.PredefinedFuncRegistry
 import sigmastate.lang.syntax.ParserException
 
 /**
-  * @param networkPrefix network prefix to decode an ergo address from string (PK op)
-  * @param builder
+  * @param networkPrefix    network prefix to decode an ergo address from string (PK op)
+  * @param builder          used to create ErgoTree nodes
+  * @param lowerMethodCalls if true, then MethodCall nodes are lowered to ErgoTree nodes
+  *                         when [[sigmastate.SMethod.irInfo.irBuilder]] is defined
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/lang/SigmaCompiler.scala:17]: What does that mean for a user ?

File: sigmastate/src/main/scala/sigmastate/utxo/transformers.scala [hunk 4]
```diff
@@ -9,12 +9,27 @@ import sigmastate.serialization.OpCodes
 import org.ergoplatform.ErgoBox.RegisterId
 import sigmastate.Operations._
 import sigmastate.lang.exceptions.InterpreterException
+import special.sigma.Box
 
 
+// TODO refactor: remove this trait as it doesn't have semantic meaning
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/utxo/transformers.scala:15]: What does have then? Or what does the TODO mean if no replacement?

**Review by kushti — APPROVED** [review]
LGTM but please check comments

File: sigmastate/src/main/scala/sigmastate/interpreter/CryptoFunctions.scala (hunk 1 above)
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/interpreter/CryptoFunctions.scala:8]: added detailed description

File: sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala (hunk 2 above)
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala:152]: I think no harm to be explicit. Updated comment.

File: sigmastate/src/main/scala/sigmastate/lang/SigmaCompiler.scala (hunk 3 above)
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/lang/SigmaCompiler.scala:17]: Answered in the updated description.

File: sigmastate/src/main/scala/sigmastate/utxo/transformers.scala (hunk 4 above)
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/utxo/transformers.scala:15]: removed this TODO

**Pull request merged**